docs/types.rst

```diff
@@ -133,20 +133,31 @@ As a rule of thumb, use `bytes` for arbitrary-length raw byte data and `string`
 for arbitrary-length string (utf-8) data. If you can limit the length to a certain
 number of bytes, always use one of `bytes1` to `bytes32` because they are much cheaper.
 
-.. index:: literal, literal;integer
+.. index:: ! ufixed, ! fixed, ! fixed point number
 
-.. _integer_literals:
+Fixed Point Numbers
+-------------------
 
-Integer Literals
------------------
+**bold** COMING SOON... **bold**
 
-Integer Literals are arbitrary precision integers until they are used together with a non-literal. In `var x = 1 - 2;`, for example, the value of `1 - 2` is `-1`, which is assigned to `x` and thus `x` receives the type `int8` -- the smallest type that contains `-1`, although the natural types of `1` and `2` are actually `uint8`.    
 
-It is even possible to temporarily exceed the maximum of 256 bits as long as only integer literals are used for the computation: `var x = (0xffffffffffffffffffff * 0xffffffffffffffffffff) * 0;` Here, `x` will have the value `0` and thus the type `uint8`.
+.. index:: literal, literal;rational
 
-.. warning::
-    Divison on integer literals used to truncate in earlier versions, but it will actually convert into a rational number in the future, i.e. `1/2` is not equal to `0`, but to `0.5`.
+.. _rationals:
+
+Rationals
+---------
+
+Rationals are a fraction composed of two arbitrary precision integers until they are used together with a non-literal. In `var x = 1 - 2;`, for example, the value of `1 - 2` is `-1/1`, which is assigned to `x` and thus `x` receives the type `int8` -- the smallest type that contains `-1`, although the natural types of `1` and `2` are actually `uint8`.
+
+This quality gives rationals the ability to very precisely represent fixed point types as well as integer types. Whenever a decimal is detected, it is immediately turned into a a precise fractional representation of the decimal number. For example, the value `0.5` is represented as `1/2` and vice versa. When converted to fixed point types, these numbers truncate after a division creates one big integer.
 
+One of the great additions of this feature is the capability to now exponentiate by negative integers. `3 ** -2` now produces the value `1/9`. However, it is not possible to reliably calculate the exponent of a rational so they are disallowed. Fractional numbers also cannot use bit operations such as `BitOr(|)`, `BitXor(^)`, `BitNot(~)` and `BitAnd(&)`.
+
+It is even possible to temporarily exceed the maximum of 256 bits as long as only rationals are used for the computation: `var x = (0xffffffffffffffffffff * 0xffffffffffffffffffff) * 0;` Here, `x` will have the value `0` and thus the type `uint8`.
+
+.. warning::
+    While this type is available to use, fractional numbers are not currently implicitly convertable to integer types. Even if the result of the expression is an integer, the type will default to the yet to be released fixed point type
```

Comment:
but it is true....remember...we tested this...we said that we are going to take care of this in a later refactor.